content/docs/examples/advanced-gateways/ingress-certmgr/index.md

```diff
@@ -0,0 +1,174 @@
+  --set certmanager.enabled=true \
+  --set certmanager.email=mailbox@donotuseexample.com \
+  > $HOME/istio-fetch/istio.yaml
+{{< /text >}}
+
+## Configuring DNS name and Gateway
+
+Take a note of the external IP address of the `istio-ingressgateway` service:
+
+{{< text bash >}}
+$ kubectl -n istio-system get service istio-ingressgateway
+{{< /text >}}
+
+Configure your DNS zone so that the domain you'd like to use for this tutorial is resolving to the external IP address of `istio-ingressgateway` service that you've captured in the previous step. You will need a real domain name for this tutorial in order to get a TLS certificate issued. Let's store the configured domain name into an environment variable for further use:
+
+{{< text bash >}}
+$ INGRESS_DOMAIN=mysubdomain.mydomain.edu
+{{< /text >}}
+
+Your Istio installation contains an automatically generated `Gateway` resource configured to serve the routes defined by the Kubernetes `Ingress` resources. By default it does not use SDS (Secrets Discovery Service), so we need to modify it in order to enable the delivery of the TLS certificates to the `istio-ingressgateway` via SDS:
+
+{{< text bash >}}
+$ kubectl -n istio-system edit gateway
+{{< /text >}}
+
+...and modify the `tls` section corresponding to the `https-default` port as follows:
+
+{{< text bash >}}
+$ kubectl -n istio-system \
+  patch gateway istio-autogenerated-k8s-ingress --type=json \
+  -p='[{"op": "replace", "path": "/spec/servers/1/tls", "value": {"credentialName": "ingress-cert-staging", "mode": "SIMPLE", "privateKey": "sds", "serverCertificate": "sds"}}]'
+{{< /text >}}
+
+Now it's time to setup a demo application.
+
+## Setting up a demo application
+
+We will be using a simple `helloworld` application for this tutorial. The following command will spin up the `Deployment` and `Service` for the demo application and expose the service using an `Ingress` resource that will be handled by `istio-ingressgateway`.
+
+{{< text bash >}}
+$ cat <<EOF | kubectl apply -f -
+apiVersion: v1
+kind: Service
+metadata:
+  name: helloworld
+  labels:
+    app: helloworld
+spec:
+  ports:
+  - port: 5000
+    name: http
+  selector:
+    app: helloworld
+---
+apiVersion: extensions/v1beta1
+kind: Deployment
+metadata:
+  name: helloworld
+spec:
+  template:
+    metadata:
+      labels:
+        app: helloworld
+    spec:
+      containers:
+      - name: helloworld
+        image: istio/examples-helloworld-v1
+        resources:
+          requests:
+            cpu: "100m"
+        imagePullPolicy: IfNotPresent
+        ports:
+        - containerPort: 5000
+---
+apiVersion: extensions/v1beta1
+kind: Ingress
+metadata:
+  annotations:
+    kubernetes.io/ingress.class: istio
+  name: helloworld-ingress
+spec:
+  rules:
+    - host: "$INGRESS_DOMAIN"
+      http:
+        paths:
+          - path: /hello
+            backend:
+              serviceName: helloworld
+              servicePort: 5000
+---
```

Comment:
aren't we supposed to configure TLS here to access the host?  I'm a little surprised it is only http here.

Reply:
Ingress support is not feature complete. TLS configuration is only supported via the Gateway that handles Ingress resources (that's what we're patching earlier). We have been asked about this feature, but I'm not sure what is the priority atm.